packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -2977,9 +3028,22 @@ AND event_name = '${eventName}'`,
           eventColumn: "event",
           timestampColumn: "received_at",
           userIdColumn: "user_id",
+          filterColumns: `
+          (CASE
+            WHEN context_user_agent LIKE '%Mobile%' THEN 'Mobile'
+            ELSE 'Tablet/Desktop' END
+          ) as device,
+          (CASE 
+            WHEN context_user_agent LIKE '% Firefox%' THEN 'Firefox'
+            WHEN context_user_agent LIKE '% OPR%' THEN 'Opera'
+            WHEN context_user_agent LIKE '% Edg%' THEN ' Edge' 
+            WHEN context_user_agent LIKE '% Chrome%' THEN 'Chrome'
+            WHEN context_user_agent LIKE '% Safari%' THEN 'Safari'
+            ELSE 'Other' END
+          ) as browser`,
```

Comment:
This is missing the following for segment, although it looks like Rudderstack does not support these.  So maybe we need to treat them separately here instead of in a single case block

```sql
  context_campaign_source as source,
  context_campaign_medium as medium,
```

Reply:
@jdorn Hmm... neither of my Segment or Rudderstack sandboxes seem to be tracking these columns.

Do you recall where you saw documentation around this?

I'm looking at this article for [Segment](https://segment.com/docs/connections/storage/warehouses/schema/#event-tables) and this [article](https://www.rudderstack.com/docs/destinations/warehouse-destinations/warehouse-schema/) for Rudderstack.

Note: I do see this [article](https://www.rudderstack.com/docs/event-spec/standard-events/common-fields/) for Rudderstack showing the available contextual fields, but it doesn't appear that they're automatic.

Reply:
Per discussion, it appears that segment will only add the above columns if they're tracked, so we're not including them here.